qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/IndexingIT.java

```diff
@@ -5,27 +5,123 @@
+    public void testKnnIndexDefaultLegacyFieldMapping() throws Exception {
         waitForClusterHealthGreen(NODES_BWC_CLUSTER);
 
         if (isRunningAgainstOldCluster()) {
             createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
-            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_CNT);
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_COUNT);
+        }
+
+        else {
+            kNNIndexUpgradedCluster();
+        }
+    }
+
+    // Custom Legacy Field Mapping
+    // space_type : "linf", engine : "nmslib", m : 2, ef_construction : 2
+    public void testKnnIndexCustomLegacyFieldMapping() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(
+                testIndex,
+                getKNNIndexCustomLegacyFieldMappingSettings(
+                    SpaceType.LINF,
+                    KNN_ALGO_PARAM_M_MIN_VALUE,
+                    KNN_ALGO_PARAM_EF_CONSTRUCTION_MIN_VALUE
+                ),
+                createKnnIndexMapping(TEST_FIELD, DIMENSIONS)
+            );
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_COUNT);
+        } else {
+            kNNIndexUpgradedCluster();
+        }
+    }
+
+    // Default Method Field Mapping
+    // space_type : "l2", engine : "nmslib", m : 16, ef_construction : 512
+    public void testKnnIndexDefaultMethodFieldMapping() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMethodFieldMapping(TEST_FIELD, DIMENSIONS));
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_COUNT);
+        } else {
+            kNNIndexUpgradedCluster();
+        }
+    }
+
+    // Custom Method Field Mapping
+    // space_type : "inner_product", engine : "faiss", m : 50, ef_construction : 1024
+    public void testKnnIndexCustomMethodFieldMapping() throws Exception {
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(
+                testIndex,
+                getKNNDefaultIndexSettings(),
+                createKnnIndexCustomMethodFieldMapping(TEST_FIELD, DIMENSIONS, SpaceType.INNER_PRODUCT, KNN_ENGINE_FAISS, 50, 1024)
+            );
+            addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 0, ADD_DOCS_COUNT);
+        } else {
+            kNNIndexUpgradedCluster();
+        }
+    }
+
+    public void kNNIndexUpgradedCluster() throws Exception {
+        validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 10, K);
+        cleanUpCache();
+        addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, 10, ADD_DOCS_COUNT);
+        validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 20, K);
+        forceMergeKnnIndex(testIndex);
+        validateKNNSearch(testIndex, TEST_FIELD, DIMENSIONS, 20, K);
+        deleteKNNIndex(testIndex);
+    }
+
+    public void testNullParametersOnUpgrade() throws Exception {
```

Comment:
Could you add 2 similar tests: empty parameters and then also when there are no parameters?

Reply:
added